airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -117,5 +116,16 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
 
         :param config: A Mapping of the user input configuration as defined in the connector spec.
         """
-        pass
+        self._auth = BigqueryAuth(config)
+
+        if not self.streams_catalog:
+            self.discover(None, config)
+        for stream in self.streams_catalog:
+            yield BigqueryStream(
```

Comment:
The return type is a list and not a generator here

Reply:
Ok I will update it

Reply:
You can't just update the typing as this inherits from a class that expose the abstract method `def streams(<...>) -> List[Stream]` (see [here](https://github.com/airbytehq/airbyte/blob/5b64af09b4d746b92a0e6c7779a4f886f21777ec/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L59)). Hence, the return type must align with the parent interface